packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -156,6 +156,41 @@ const ResizableTitle: React.FC<ResizableProps & ResizableState> = (props) => {
   );
 };
 
+const CustomRowWithStyles: React.FC<
```

Comment:
You can just use `memo` around the component definition since a lot of your hooks are dependent on the props.